# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/loader/ModuleLoaderBase.cpp
# Commit: a80062ebc58c
# Full Hash: a80062ebc58cb7052b0cba86f7503b84cda49f9a
# Author: Yoshi Cheng-Hao Huang <allstars.chh@gmail.com>
# Date: 2025-10-03 20:53 +0000
# Regressor Bug: 1968890
# File Overlap Count: 2
# Overlapping Files: js/loader/ModuleLoaderBase.h, js/loader/ModuleLoaderBase.cpp
# Description:
#   Bug 1968890 - Part 8: Remove FinishDynamicImportAndReject. r=jonco
# ==============================================================================

diff -r f0d873f64979 -r a80062ebc58c js/loader/ModuleLoaderBase.cpp
--- a/js/loader/ModuleLoaderBase.cpp	Fri Oct 03 20:53:29 2025 +0000
+++ b/js/loader/ModuleLoaderBase.cpp	Fri Oct 03 20:53:29 2025 +0000
@@ -1411,37 +1411,6 @@
   mDynamicImportRequests.AppendElement(aRequest);
 }
 
-void ModuleLoaderBase::FinishDynamicImportAndReject(ModuleLoadRequest* aRequest,
-                                                    nsresult aResult) {
-  AutoJSAPI jsapi;
-  MOZ_ASSERT(NS_FAILED(aResult));
-  if (!jsapi.Init(mGlobalObject)) {
-    return;
-  }
-
-  if (aRequest->mPayload.isUndefined()) {
-    // Import has already been completed.
-    return;
-  }
-
-  JSContext* cx = jsapi.cx();
-  Rooted<Value> payload(cx, aRequest->mPayload);
-
-  if (NS_FAILED(aResult) &&
-      aResult != NS_SUCCESS_DOM_SCRIPT_EVALUATION_THREW_UNCATCHABLE) {
-    MOZ_ASSERT(!JS_IsExceptionPending(cx));
-    nsAutoCString url;
-    aRequest->mURI->GetSpec(url);
-    JS_ReportErrorNumberASCII(cx, js::GetErrorMessage, nullptr,
-                              JSMSG_DYNAMIC_IMPORT_FAILED, url.get());
-    FinishLoadingImportedModuleFailedWithPendingException(cx, payload);
-  } else {
-    FinishLoadingImportedModuleFailed(cx, payload, UndefinedHandleValue);
-  }
-
-  aRequest->ClearImport();
-}
-
 ModuleLoaderBase::ModuleLoaderBase(ScriptLoaderInterface* aLoader,
                                    nsIGlobalObject* aGlobalObject)
     : mGlobalObject(aGlobalObject), mLoader(aLoader) {
@@ -1501,7 +1470,6 @@
   return !mDynamicImportRequests.isEmpty();
 }
 
-// TODO: Bug 1968890 : Update error handling for dynamic import
 void ModuleLoaderBase::CancelDynamicImport(ModuleLoadRequest* aRequest,
                                            nsresult aResult) {
   // aRequest may have already been unlinked by CC.
@@ -1512,12 +1480,7 @@
     // If the ClearDynamicImport() has been called, then it should have been
     // removed from mDynamicImportRequests as well.
     MOZ_ASSERT(!aRequest->mPayload.isUndefined());
-
     aRequest->Cancel();
-    // FinishDynamicImport must happen exactly once for each dynamic import
-    // request. If the load is aborted we do it when we remove the request
-    // from mDynamicImportRequests.
-    FinishDynamicImportAndReject(aRequest, aResult);
   }
 }
 